docs/structured-streaming-kafka-integration.md

```diff
@@ -624,3 +624,56 @@ For experimenting on `spark-shell`, you can also use `--packages` to add `spark-
 
 See [Application Submission Guide](submitting-applications.html) for more details about submitting
 applications with external dependencies.
+
+## Security
+
+Kafka 0.9.0.0 introduced several features that increases security in a cluster. For detailed
+description about these possibilities, see [Kafka security docs](http://kafka.apache.org/documentation.html#security).
+
+It's worth noting that security is optional and turned off by default.
+
+Spark supports the following ways to authenticate against Kafka cluster:
+- **Delegation token (introduced in Kafka broker 1.1.0)**: This way the application can be configured
+  via Spark parameters and may not need JAAS login configuration (Spark can use Kafka's dynamic JAAS
+  configuration feature). For further information about delegation tokens, see
+  [Kafka delegation token docs](http://kafka.apache.org/documentation/#security_delegation_token).
+
+  The process is initiated by Spark's Kafka delegation token provider. This is enabled by default
```

Comment:
I think this needs to be reworded, since "enabled by default" is misleading. You're required to set `spark.kafka.bootstrap.servers`  before this actually works. The `enabled` setting is only really useful to disable DTs when the other config has been set (e.g. if it's set in a shared config file and someone wants to disable it).

Reply:
Fixed.